src/compiler/msvc.rs

```diff
@@ -208,88 +246,48 @@ pub fn parse_arguments(arguments: &[OsString]) -> CompilerArguments<ParsedArgume
         }
     });
 
-    // Next split off arguments with a value, creating an iterator of tuples
-    let mut it = it.map(|arg| {
-        if let Some(arg) = arg.split_prefix("-Fo") {
-            ("-Fo".into(), Some(arg))
-        } else if let Some(arg) = arg.split_prefix("-deps") {
-            ("-deps".into(), Some(arg))
-        } else if let Some(p) = arg.split_prefix("-Fd") {
-            ("-Fd".into(), Some(p))
-        } else {
-            (arg, None)
-        }
-    });
-
-    while let Some((flag, mut arg)) = it.next() {
-        if let Some(s) = flag.to_str() {
-            let mut handled = true;
-            match s {
-                "-c" => compilation = true,
-                "-FI" => {
-                    common_args.push("-FI".into());
-                    if let Some((arg_val, flag)) = it.next() {
-                        if flag.is_some() {
-                            return CompilerArguments::CannotCache("extra -FI arg")
-                        }
-                        common_args.push(arg_val);
+    for item in ArgsIter::new(it, &ARGS[..]) {
+        match item.data {
+            Some(TooHard) => {
+                return CompilerArguments::CannotCache(item.arg.to_str().expect(
```

Comment:
You could make `TooHard` contain a `&'static str` and just return that in the `CannotCache`, but it's not particularly pressing.